crates/bevy_ecs/src/component.rs

```diff
@@ -1133,3 +1145,131 @@ impl<T: Component> FromWorld for InitComponentId<T> {
         }
     }
 }
+#[cfg(feature = "track_change_detection")]
+type RequiredComponentConstructor =
+    dyn Fn(&mut Table, &mut SparseSets, Tick, TableRow, Entity, &'static Location<'static>);
+#[cfg(not(feature = "track_change_detection"))]
+type RequiredComponentConstructor = dyn Fn(&mut Table, &mut SparseSets, Tick, TableRow, Entity);
+
+/// A context-less constructor that can be called to construct a new instance of a component.
```

Comment:
These docs are quite odd to me; they don't actual mention / tie into the notion of what a required component *is*, and instead focus purely on the mechanics.